sky/sdk/lib/widgets/tabs.dart

```diff
@@ -345,32 +347,73 @@ class Tab extends Component {
   }
 }
 
+class _TabsScrollSimulation extends FrictionSimulation {
+  _TabsScrollSimulation(
+    double position,
+    double velocity,
+    double this._minScrollOffset,
+    double this._maxScrollOffset) : super(0.025, position, velocity);
+
+  final double _minScrollOffset;
+  final double _maxScrollOffset;
+
+  double x(double time) {
+    return super.x(time).clamp(_minScrollOffset, _maxScrollOffset);
+  }
+
+  bool isDone(double time) {
+    return super.isDone(time) ||
+      (x(time) - _minScrollOffset).abs() < tolerance.distance ||
+      (x(time) - _maxScrollOffset).abs() < tolerance.distance;
+  }
+}
```

Comment:
I've renamed it.

I suppose it belongs in Newton. That will take two steps: get it uulled into Newton and then update DEPS.